cmd/spire-server/cli/agent/list.go

```diff
@@ -0,0 +1,99 @@
+package agent
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+
+	"github.com/spiffe/spire/cmd/spire-server/util"
+	"github.com/spiffe/spire/proto/api/registration"
+	"github.com/spiffe/spire/proto/common"
+
+	"golang.org/x/net/context"
+)
+
+//ListConfig holds configuration for ListCLI
+type ListConfig struct {
+	// Socket path of registration API
+	RegistrationUDSPath string
+}
+
+// Validate will perform a basic validation on config fields
+func (c *ListConfig) Validate() (err error) {
+	if c.RegistrationUDSPath == "" {
+		return errors.New("a socket path for registration api is required")
+	}
+	return nil
+}
+
+//ListCLI command for listing attested nodes
+type ListCLI struct {
+	RegistrationClient registration.RegistrationClient
+	NodeList           []*common.AttestedNode
+}
+
+func (ListCLI) Synopsis() string {
+	return "List agent spiffeIDs"
+}
+
+func (c ListCLI) Help() string {
+	_, err := c.parseConfig([]string{"-h"})
+	return err.Error()
+}
+
+//Run will lists attested agents
+func (c *ListCLI) Run(args []string) int {
+	ctx := context.Background()
+
+	config, err := c.parseConfig(args)
+	if err != nil {
+		fmt.Println(err.Error())
+		return 1
+	}
+
+	if err = config.Validate(); err != nil {
+		fmt.Println(err.Error())
+		return 1
+	}
+
+	if c.RegistrationClient == nil {
+		c.RegistrationClient, err = util.NewRegistrationClient(config.RegistrationUDSPath)
+		if err != nil {
+			fmt.Printf("Error creating registration client: %v \n", err)
+			return 1
+		}
+	}
+
+	listResponse, err := c.RegistrationClient.ListAgents(ctx, &common.Empty{})
+	if err != nil {
+		fmt.Printf("Error listing attested nodes: %v \n", err)
+		return 1
+	}
+	c.NodeList = listResponse.Nodes
```

Comment:
IMO it will be cleaner to pass this list directly to `printAttestedNodes` function rather than first setting it on the struct and then looking it up again.

Reply:
I thought the same, but this list needs to be accessed from the test suite to do some assertions... that is why `NodeList` is set as an attribute of `ListCLI`